x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/index_migrator/index.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient, Logger } from '@kbn/core/server';
+import { RuleMigrationSpaceIndexMigrator } from './per_space';
+import type { Adapters } from '../types';
+
+export class RuleMigrationIndexMigrator {
+  constructor(
+    private ruleMigrationIndexAdapters: Adapters,
+    private esClient: ElasticsearchClient,
+    private logger: Logger
+  ) {}
+
+  private async getSpaceListForMigrations() {
+    const rulesIndexPattern = this.ruleMigrationIndexAdapters.rules.name;
+    const rulesIndicesAcrossSpaces = await this.esClient.indices.get({
+      index: rulesIndexPattern,
+      allow_no_indices: true,
+    });
+
+    const spaceList = Object.keys(rulesIndicesAcrossSpaces).map((index) => {
+      const indexPatternPrefix = rulesIndexPattern.replace('*', '');
+      return index.replace(`${indexPatternPrefix}`, '');
+    });
+
+    return spaceList;
```

Comment:
This logic is not entirely correct, since the name is just the prefix, it does not include the `-<spaceId>` part, to do so we need to use the `getIndexName` function:

```suggestion
    const rulesIndicesAcrossSpaces = await this.esClient.indices.get({
      index: this.ruleMigrationIndexAdapters.rules.getIndexName('*'),
      allow_no_indices: true,
    });

    const rulesIndexPrefix = this.ruleMigrationIndexAdapters.rules.getIndexName('');
    const spaceList = Object.keys(rulesIndicesAcrossSpaces).map((index) => {
      return index.replace(rulesIndexPrefix, '');
    });

    return spaceList;
```

Reply:
Let me know if you still maintain this opinion after the above comment I posted.

Reply:
Answered above. 
You are right, but yes, I would still use `getIndexName` instead to make the logic explicit.